doc/api/n-api.md

```diff
@@ -0,0 +1,3150 @@
+It's used to wrap native functions so that they can be invoked from JavaScript.
+
+JavaScript Functions are described in 
+[Section 19.2](https://tc39.github.io/ecma262/#sec-function-objects) 
+of the ECMAScript Language Specification.
+
+#### *napi_create_object*
+
+##### Signature
+```C
+napi_status napi_create_object(napi_env env, napi_value* result)
+```
+
+##### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[out] result`: A `napi_value` representing a JavaScript Object
+
+##### Return value
+- `napi_ok` if the API succeeded.
+
+##### Description
+This API allocates a default JavaScript Object. 
+It is the equivalent of doing `new Object()` in JavaScript.
+
+The JavaScript Object type is described in 
+[Section 6.1.7](https://tc39.github.io/ecma262/#sec-object-type) of the 
+ECMAScript Language Specification.
+
+#### *napi_create_symbol*
+
+##### Signature
+```C
+napi_status napi_create_symbol(napi_env env,
+                               const char* description,
+                               napi_value* result)
+```
+
+##### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  description`: Null-terminated character buffer representing a 
+UTF8-encoded string to describe the symbol
+- `[out] result`: A `napi_value` representing a JavaScript Symbol
+
+##### Return value
+- `napi_ok` if the API succeeded.
+
+##### Description
+This API creates a JavaScript Symbol object from a UTF8-encoded C++ string
+
+The JavaScript Symbol type is described in 
+[Section 19.4](https://tc39.github.io/ecma262/#sec-symbol-objects) 
+of the ECMAScript Language Specification.
+
+#### *napi_create_typedarray*
+
+##### Signature
+```C
+napi_status napi_create_typedarray(napi_env env,
+                                   napi_typedarray_type type,
+                                   size_t length,
+                                   napi_value arraybuffer,
+                                   size_t byte_offset,
+                                   napi_value* result)
+```
+
+##### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  type`: Scalar datatype of the elements within the TypedArray
+- `[in]  length`: Number of elements in the TypedArray
+- `[in]  arraybuffer`: ArrayBuffer underlying the typed array
+- `[in]  byte_offset`: The byte offset within the ArrayBuffer from which to
+start projecting the TypedArray
+- `[out] result`: A `napi_value` representing a JavaScript TypedArray
+
+##### Return value
+- `napi_ok` if the API succeeded.
+
+##### Description
+This API creates a JavaScript TypedArray object over an existing ArrayBuffer.
+TypedArray objects provide an array-like view over an underlying data buffer 
+where each element has the same underlying binary scalar datatype.
+
+It's required that (length * size_of_element) + byte_offset should 
+be <= ArrayBuffer.ArrayBufferByteLength. If not, a RangeError exception is 
```

Comment:
What is ArrayBuffer.ArrayBufferByteLength? Maybe this should read sth like `arraybuffer.byteLength`, otherwise it looks like a static property of the ArrayBuffer constructor itself